packages/bento-design-system/src/Charts/Legend/Legend.tsx

```diff
@@ -0,0 +1,43 @@
+import { LegendProps as RechartsLegendProps } from "recharts";
+import type { Payload } from "recharts/types/component/DefaultLegendContent";
+import { match } from "ts-pattern";
+import { Box } from "../../Box/Box";
+import { Column, Columns } from "../../Layout/Columns";
+import { Inline } from "../../Layout/Inline";
+import { Body } from "../../Typography/Body/Body";
+
+export const legendContent: Required<RechartsLegendProps>["content"] = ({ payload = [] }) => {
+  return (
+    <Box paddingTop={8} width="full">
+      <Inline space={16} alignY="center" align="center">
+        {payload.map((entry) =>
+          match(entry.type ?? "square")
+            .with("plainline", "line", () => <LegendItemLine key={entry.value} {...entry} />)
+            .otherwise(() => <LegendItemArea key={entry.value} {...entry} />)
```

Comment:
are there many other possible values? Can't we use an exhaustive match instead?

Reply:
there are many others, but not too many, but I can make this exhaustive.